### PR TITLE
#493 Include context in slack extra info.

### PR DESCRIPTION
**Pull request by jamiel**
Adds the context items to Slack attachments.

```
$handler->addError('Foo', array('foo' => 'bar'));
```
screen shot 2015-02-09 at 22 42 36

### CODE REVIEW

File: src/Monolog/Handler/SlackHandler.php [hunk 1]
```diff
@@ -169,6 +169,15 @@ private function buildContent($record)
...
                 );
+
+                // Add all context items as individual fields
+                foreach ($record['context'] as $var => $val) {
+                    $attachment['fields'][] = array(
+                        'title' => $var,
+                        'value' => $val,
+                        'short' => true
+                    );
+                }
```
Comment (Seldaek) [inline src/Monolog/Handler/SlackHandler.php]: I am ok with this but I think it should be in line with the Extra stuff, and both should actually follow the useShortAttachment flag. If it's on both context and extra should be like lines 161-171 above, and if it's false both should be like you have it for the context. And if extra|context are empty no attachment should be created IMO, even in useShortAttachment mode.

**Comment by noetix** [comment]
+1

**Comment by jamiel** [comment]
That's been adjusted so that both extra and context are consistent depending on useShortAttachment and are only added if the fields are present and not empty.

To be honest, it's still not perfect as you should be able to choose one or the other and short attachment shouldn't be used to decide whether to stringify the the object or not as Slack's intended use is just whether the fields can be stacked next to each other not in two columns.

**Comment by Seldaek** [comment]
Thanks! That sounds like good progress already so I'll merge, but if you feel like this can be improved upon please send another PR. I'd just like to see before/after screenshots because I don't really have the time or ability to test with slack to see how it comes out.

**Pull request merged**